Sources/ManageErrors.php

```diff
@@ -180,7 +180,7 @@ function ViewErrorLog()
 			'time' => timeformat($row['log_time']),
 			'timestamp' => $row['log_time'],
 			'url' => array(
-				'html' => $smcFunc['htmlspecialchars'](strpos($row['url'], 'cron.php') === false ? (substr($row['url'], 0, 1) == '?' ? $scripturl : '') . $row['url'] : $row['url']),
+				'html' => $smcFunc['htmlspecialchars'](strpos($row['url'], 'cron.php') === false && !strpos($row['url'], 'https://') && !strpos($row['url'], 'http://')  ? (substr($row['url'], 0, 1) == '?' ? $scripturl : '') . $row['url'] : $row['url']),
```

Comment:
```suggestion
				'html' => $smcFunc['htmlspecialchars'](strpos($row['url'], 'cron.php') === false && !strpos($row['url'], 'https://') && !strpos($row['url'], 'http://')  ? ($row['url'][0] === '?' ? $scripturl : '') . $row['url'] : $row['url']),
```
perf: Check string inline; do not check copy

Reply:
How's that for premature optimization, lol

Reply:
Looks good.  I didn't even think about it using more memory that way as its copying it.  Really wish PHP had kept {} as the way to navigate strings by character offset.  So confusing think its an array.  Yes I know technically in C/C++ a string is an array of chars, but still...

Reply:
I sent up a change for the strpos stuff I did.  http:// could exist somewhere in the string and break this.  So I check it starts with 0 and then flip the check. 

```
<?php
$url = 'https://google.com';

var_dump(strpos($url, 'cron.php') === false);
var_dump(!strpos($url, 'https://'));
var_dump(!strpos($url, 'http://'));

$url = 'bad-actor/https://google.com';

var_dump(strpos($url, 'cron.php') === false);
var_dump(!(strpos($url, 'https://') === 0));
var_dump(!(strpos($url, 'http://') === 0));
```